pumpkin-world/src/block/interactive/sign.rs

```diff
@@ -0,0 +1,146 @@
+use pumpkin_core::math::position::WorldPosition;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Copy)]
+#[repr(u16)]
+pub enum SignType {
+    OakSign = 194,
+    SpruceSign = 195,
+    BirchSign = 196,
+    AcaciaSign = 197,
+    CherrySign = 198,
+    JungleSign = 199,
+    DarkOakSign = 200,
+    PaleOakSign = 201,
+    MangroveSign = 202,
+    BambooSign = 203,
+
+    OakWallSign = 208,
+    SpruceWallSign = 209,
+    BirchWallSign = 210,
+    AcaciaWallSign = 211,
+    CherryWallSign = 212,
+    JungleWallSign = 213,
+    DarkOakWallSign = 214,
+    PaleOakWallSign = 215,
+    MangroveWallSign = 216,
+    BambooWallSign = 217,
+
+    OakHangingSign = 218,
+    SpruceHangingSign = 219,
+    BirchHangingSign = 220,
+    AcaciaHangingSign = 221,
+    CherryHangingSign = 222,
+    JungleHangingSign = 223,
+    DarkOakHangingSign = 224,
+    PaleOakHangingSign = 225,
+    CrimsonHangingSign = 226,
+    WarpedHangingSign = 227,
+    MangroveHangingSign = 228,
+    BambooHangingSign = 229,
+
+    OakWallHangingSign = 230,
+    SprucWalleHangingSign = 231,
+    BirchWallHangingSign = 232,
+    AcaciaWallHangingSign = 233,
+    CherryWallHangingSign = 234,
+    JungleWallHangingSign = 235,
+    DarkOakWallHangingSign = 236,
+    PaleOakWallHangingSign = 237,
+    CrimsonWallHangingSign = 238,
+    WarpedWallHangingSign = 239,
+    MangroveWallHangingSign = 240,
+    BambooWallHangingSign = 241,
+
+    CrimsonSign = 849,
+    WarpedSign = 850,
+    CrimsonWallSign = 851,
+    WarpedWallSign = 852,
+}
```

Comment:
This looks horrible, Also the the ID's will change with every Update

Reply:
Now looking at it again, it is. I will make sure to fix this using the block.json.

Reply:
Added a macro that looks up the block entity id instead of the actual id and eliminated the mess.